pkg/core/managers/apis/dataplane/dataplane_manager.go

```diff
@@ -55,10 +56,15 @@ func (m *dataplaneManager) Create(ctx context.Context, resource core_model.Resou
 	m.setInboundsClusterTag(dp)
 	m.setGatewayClusterTag(dp)
 	m.setHealth(dp)
+	existingLabels := map[string]string{}
+	maps.Copy(existingLabels, opts.Labels)
+	if resource.GetMeta() != nil && resource.GetMeta().GetLabels() != nil {
```

Comment:
Something is odd here, I don't think we expect `resource.GetMeta()` to be set in the `Create` method, no?

Reply:
It is set in case of creating dp. Because we are doing upsert on dp creation: https://github.com/kumahq/kuma/blob/72bc7f18f2f6cbb14816e2477246dd3191fede7e/pkg/xds/server/callbacks/dataplane_lifecycle.go#L123-L129

Reply:
It's set only in this PR with `existing.SetMeta(md.Resource.GetMeta())`, I don't think we've done it before. Normally, I don't think you can set meta when calling `Create` because ResourceStores are the only ones who can create store-specific meta.